openapi.yaml

```diff
@@ -1,190 +1,480 @@
+          description: Filter parks by owner ID
+          required: false
+          type: string
+        - in: query
+          name: city
+          description: Filter parks by city
+          required: false
+          type: string
+        - in: query
+          name: state
+          description: Filter parks by state
           required: false
           type: string
+          pattern: '^(?:(A[KLRZ]|C[AOT]|D[CE]|FL|GA|HI|I[ADLN]|K[SY]|LA|M[ADEINOST]|N[CDEHJMVY]|O[HKR]|P[AR]|RI|S[CD]|T[NX]|UT|V[AIT]|W[AIVY]))$'
+        - in: query
+          name: zipCode
+          description: Filter parks by zip code
+          required: false
+          type: integer
       responses:
         '200':
-          description: Successful response
+          description: Successful Response
+          schema:
+            $ref: '#/definitions/ParksResult'
+        '400':
+          $ref: '#/responses/400'
+        '500':
+          $ref: '#/responses/500'
+    post:
+      summary: Creates a new park
+      tags:
+        - parks
+      operationId: createPark
+      parameters:
+      - $ref: '#/parameters/parkPostBody'
+      responses:
+        '201':
+          description: Successfully created new park
+          headers:
+            Location:
+              type: string
+              description: Location of newly created park
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/ParkResult'
         '400':
-          description: Bad request
+          $ref: '#/responses/400'
+        '404':
+          description: Referenced owner does not exist
+          schema:
+            $ref: '#/definitions/ErrorResult'
+        '409':
+          description: Conflict with type
           schema:
             $ref: '#/definitions/ErrorResult'
         '500':
-          description: Internal server error
+          $ref: '#/responses/500'
+  /parks/{parkId}:
+    get:
+      summary: Lists info for a specific park
+      tags:
+        - parks
+      description: Get park by unique ID
+      operationId: getParkById
+      parameters:
+        - $ref: '#/parameters/parkIdPath'
+      responses:
+        '200':
+          description: Successful Response
+          schema:
+            $ref: '#/definitions/ParkResult'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+    patch:
+      summary: Update information of a specific park
+      tags: 
+        - parks
+      description: Update park by unique ID
+      operationId: updateParkById
+      parameters:
+        - $ref: '#/parameters/parkIdPath'
+        - $ref: '#/parameters/parkPatchBody'
+      responses:
+        '200':
+          description: Successful Response
+          schema:
+            $ref: '#/definitions/ParkResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          description: Invalid type, must be park
```

Comment:
The 409 response for the PATCH endpoint could be either due to an invalid type or an id that doesn't match the id in the path. Could you add this to the description?